ui/src/views/compute/DeployVM.vue

```diff
@@ -1659,7 +1659,7 @@ export default {
             }
           }
         }
-        if (this.securitygroupids.length > 0) {
+        if (this.securitygroupids && this.securitygroupids.length > 0) {
```

Comment:
question: if above the update method adds an empty list, is this extra check still needed? (or vice versa?)

Reply:
I think it would be better to change it to `if (this.securitygroupids.length > 0) {`